PowerValidatedSolutions.psm1

```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
+
+    #>
+
+	Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$id,
+		[Parameter (Mandatory = $true)] [ValidateSet('vcenter', 'nsxt')] [ValidateNotNullOrEmpty()] [String]$dataSourceType
+    )
+
+    Try {
+        if ($ariaNetworksAppliance) {
+            if ($dataSourceType -eq 'vcenter') {
+                $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/vcenters/$id"
+                (Invoke-RestMethod -Method 'DEL' -Uri $uri -Headers $ariaNetworksHeader).results
+            } elseif($dataSourceType -eq 'nsxt') {
+                $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/nsxt-managers/$id"
+                (Invoke-RestMethod -Method 'DEL' -Uri $uri -Headers $ariaNetworksHeader).results
+			} else {
+            Write-Error "Not connected to VMware Aria Operations for Networks, run Request-AriaNetworksToken and try again"
+			}
+		}
+    } Catch {
+        Write-Error $_.Exception.Message
+    }
+}
+Export-ModuleMember -Function Remove-AriaNetworksDataSource
+
+Function New-AriaNetworksvCenterDataSource {
+    <#
+        .SYNOPSIS
+        Add a new vCenter Data Source to VMware Aria Operations for Networks.
+
+        .DESCRIPTION
+        The New-AriaNetworksvCenterDataSource cmdlet allows a user to add in a vCenter Server as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
+
+        .EXAMPLE
+        New-AriaNetworksvCenterDataSource -fqdn sfo-m01-vc01.sfo.rainpole.io -username svc-inv-vsphere -password VMw@re1! -nickname "sfo-m01-vc01 - Management Domain vCenter Server" -CollectorId 15832:901:1711011916294613031 -enabled true
+        This example adds a vCenter server as a new data source in VMware Aria Operations for Networks.
+
+        .PARAMETER fqdn
+        The fqdn of the vCenter Server to add as a data source.
+
+        .PARAMETER username
+        The username to use for authentication.
+
+        .PARAMETER password
+        The password to use for authentication.
+
+        .PARAMETER nickname
+        The nickname to use for this data source in VMware Aria Operations for Networks.
+
+        .PARAMETER collectorId
+        The id of the VMware Aria Operations for Networks collector node where the vCenter data source will be connected.
+
+        .PARAMETER enabled
+        The parameter to enable the datasource.
+    #>
+
+    Param (
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+    )
+
+    Try {
```

Comment:
Any reason your not consistently using the if ($ariaNetworksAppliance) { } else { Write-Error ""} consistently

Reply:
Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/5b6f2b46e0444235d6f3d74157fc959015d63688